editor/blocks/quote/index.js

```diff
@@ -0,0 +1,44 @@
+const Editable = wp.blocks.Editable;
+const { query, html } = wp.blocks.query;
+
+wp.blocks.registerBlock( 'core/quote', {
+	title: 'Quote',
+	icon: 'format-quote',
+	category: 'quote',
+
+	attributes: {
+		value: query( 'blockquote', html( 'p' ) ),
+		citation: html( 'cite,footer' )
+	},
+
+	edit( attributes, onChange ) {
+		const { value, citation } = attributes;
+
+		// Concat strings parsed by hpq from paragraph values into one string.
+		const combinedValue = value.join( '' );
+		return (
+			<blockquote>
+				<Editable
+					value={ combinedValue }
+					onChange={ ( newValue ) => onChange( { newValue } ) } />
+				<cite>
+					<Editable
+						value={ citation }
+						onChange={ ( newValue ) => onChange( { newValue } ) } />
```

Comment:
Let me try to explain:

On the parent component rendering the blocks, we keep track of the current `state` (block list) and each time this state is updated we re-render the blocks by calling the `edit` method of each block.

When this parent component renders the block (calling edit) it provides an `onChange` callback, the block can call to update its state (its attributes), so anytime a block attribute needs to be changed, we should call `onChange({ [ attributeName ]: attributeValue })`

calling the `onChange` will update the state and triggers rendering the blocks again. This is how the unidirectional data flow of React (and similar works) 

I hope this is clear enought